### ISSUE TITLE
false negative with dangerous-default-value

### ISSUE THREAD

**Issue by dpinol** [PyCQA/pylint#7818]
### Bug description

```python
# pylint: disable=missing-module-docstring
class Clazz:
    # pylint: disable=missing-docstring, too-few-public-methods
    def __init__(
        self,
        arg: str = None,
        *,
        kk: dict = {},
    ) -> None:
        pass

```

### Command used

```shell
pylint filename.py
```

### Pylint output

```shell
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 6.67/10, +3.33)
```

### Expected behavior

4: W0102: Dangerous default value {} as argument (dangerous-default-value)

### Pylint version

```shell
pylint 2.15.6
astroid 2.12.13
Python 3.10.8 (main, Oct 18 2022, 14:58:34) [GCC 11.2.0]
```

### OS / Environment

ubuntu 22.4

**Comment by mbyrnepr2** [comment]
Thank you @dpinol it looks like a bug. Removing the `*` leads to `dangerous-default-argument` being emitted; so something not quite right there.
